core/src/main/java/org/javarosa/form/api/FormEntrySession.java

```diff
@@ -0,0 +1,214 @@
+package org.javarosa.form.api;
+
+import org.javarosa.core.model.FormIndex;
+
+import java.io.Serializable;
+import java.util.Vector;
+
+/**
+ * Records form entry actions, associating form indexes with user (string)
+ * answers.  For simplicity's sake each form index appears only once in the
+ * action list. If a question answer is updated, the original form index entry
+ * is removed, and the updated entry is added to the end.
```

Comment:
@phillipm Hey, still think this comment is wrong. Good to merge once that's fixed